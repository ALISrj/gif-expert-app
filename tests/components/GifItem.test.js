import {GifItem} from "../../src/components/index.js";
import {render, screen} from "@testing-library/react";

describe('Pruebas en GifItem', () => {

    const title = "BreakingBad";
    const url = "https://www.gif.com/";
    const id = "123123;"

    test('La página hace match con el snapshot', () => {

        const { container } = render(<GifItem url={url} title={title} id={id} />);

        expect(container).toMatchSnapshot();

    });

    test("Debe mostrar la imagen con el url y alt indicado", () => {

        render(<GifItem url={url} title={title} id={id} />);
        // screen.debug();

        // expect(screen.getByRole('img').src).toBe(url);
        // expect(screen.getByRole('img').alt).toBe(title);

        const {src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test(" Debe mostrar el título en el componente", () => {
        render(<GifItem url={url} title={title} id={id} />);
        expect( screen.getByText(title) ).toBeTruthy();
    })




})