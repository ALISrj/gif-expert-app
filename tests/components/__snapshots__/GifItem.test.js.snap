// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem La p√°gina hace match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="BreakingBad"
      src="https://www.gif.com/"
    />
    <p>
      BreakingBad
    </p>
  </div>
</div>
`;
